backend/core/src/sms/services/sms.service.ts

```diff
@@ -0,0 +1,20 @@
+import { HttpException, HttpStatus, Injectable, Scope } from "@nestjs/common"
+import { UserService } from "../../auth/services/user.service"
+import { StatusDto } from "../../shared/dto/status.dto"
+import { mapTo } from "../../shared/mapTo"
+import { SmsDto } from "../dto/sms.dto"
+import { TwilioService } from "./twilio.service"
+
+@Injectable({ scope: Scope.REQUEST })
+export class SmsService {
+  constructor(private readonly userService: UserService, private readonly twilio: TwilioService) {}
+
+  async send(dto: SmsDto): Promise<StatusDto> {
+    const user = await this.userService.findByEmail(dto.userEmail)
+    if (!user.phoneNumber) {
+      throw new HttpException("User does not have a phone number set.", HttpStatus.BAD_REQUEST)
+    }
+    await this.twilio.send(dto.body, user.phoneNumber)
```

Comment:
It looks like this returns a `MessageInstance` - is there anything interesting in there that should be included in the `StatusDto` we're returning? If not, should `TwilioService.send` return anything?

Reply:
I was using the return of Promise<MessageInstance> really only as a mechanism for Twilio to trigger exceptions, without the need to wrap stuff in try/catch.

Reply:
Gotcha, I'm wondering if it's possible that some error wouldn't cause Twilio to throw an exception but instead to populate relevant fields in the `MessageInstance` it returns? ([This](https://www.rubydoc.info/gems/twilio-ruby/Twilio/REST/Api/V2010/AccountContext/MessageInstance) looks like the API, and there are e.g. fields like error_code and error_message.) In that case, we might want to return some non-OK status in the controller?

Reply:
I added a check on MessageInstance.errorCode, and throw an HttpException in that case. I verified that errorCode is null if the message was sent successfully.